tracer/src/Datadog.Trace/Configuration/TracerSettings.cs

```diff
@@ -53,6 +53,11 @@ public TracerSettings(IConfigurationSource source)
                           // backwards compatibility for names used in the past
                           source?.GetString("DD_SERVICE_NAME");
 
+            if (AzureAppServices.Metadata.IsRelevant)
+            {
+                ServiceName = ServiceName ?? AzureAppServices.Metadata.SiteName;
```

Comment:
This will work, but we should do it in `Tracer.GetApplicationName()`. That method contains the logic to determine a default service name when one is not configured.

https://github.com/DataDog/dd-trace-dotnet/blob/90f235089df659ba965c3273636730f1760c9e57/tracer/src/Datadog.Trace/Tracer.cs#L97-L100
https://github.com/DataDog/dd-trace-dotnet/blob/90f235089df659ba965c3273636730f1760c9e57/tracer/src/Datadog.Trace/Tracer.cs#L639-L644

Reply:
Done